### ISSUE TITLE
Exception when trying to Evaluate AveragedPerceptronTrainer, LinearSvm

### ISSUE THREAD

**Issue by abgoswam** [dotnet/machinelearning#1579]
For a couple of Learners, we get an exception during `Evaluate`
- AveragedPerceptronTrainer
- LinearSvm

Exception : 

Message: System.ArgumentOutOfRangeException : Probability column 'Probability' not found
Parameter name: name

Sample : 

        [Fact]
        public void OVA_BC_AP()
        {
            string dataPath = GetDataPath("breast-cancer.txt");

            // Create a new context for ML.NET operations. It can be used for exception tracking and logging, 
            // as a catalog of available operations and as the source of randomness.
            var mlContext = new MLContext(seed: 1);
            var reader = new TextLoader(mlContext, new TextLoader.Arguments()
            {
                Column = new[]
                        {
                            new TextLoader.Column("Label", DataKind.R4, 0),
                            new TextLoader.Column("Features", DataKind.R4, new [] { new TextLoader.Range(1, 9) }),
                        }
            });

            // Data
            var data = reader.Read(GetDataPath(dataPath));

            // Pipeline
            var pipeline = new AveragedPerceptronTrainer(mlContext, "Label", "Features");

            var model = pipeline.Fit(data);
            var predictions = model.Transform(data);

            // Metrics
            var metrics = mlContext.BinaryClassification.Evaluate(predictions);
        }

**Comment by yaeldekel** [comment]
Not sure this is a bug - AveragedPerceptron does not produce calibrated models. If we don't expose calibration APIs, we should probably do that. Also, we may want to consider warning when the probability column isn't there instead of throwing.

**Comment by Zruty0** [comment]
..or call `mlContext.BinaryClassification.EvaluateNonCalibrated`. 

Exposing the calibration API is fine. I would just make a calibration estimator for that though, that trains towards one parameter. Embedding calibrator into learner seems somewhat unnecessary.

**Comment by abgoswam** [comment]
Thanks for the comments. Using `mlContext.BinaryClassification.EvaluateNonCalibrated` got me metrics I was looking for.

I have a few follow up questions, based on the comments above:

-   What do me mean by the following _"make a calibration estimator that trains towards one parameter"_  ?

- My understanding is that currently some learners have calibrator embedded (e.g. FastTree)  while other  learners do not  (e.g. AveragedPerceptron) . Is that by design ?  

- From a user perspective, is there a way to know if I should use `Evaluate`  or `EvaluateNonCalibrated` . I feel this may confuse users of ML.NET

@Zruty0 @yaeldekel

**Comment by abgoswam** [comment]
Closing this issue,, since we verified  that `mlContext.BinaryClassification.EvaluateNonCalibrated` gave us the desired metrics for AveragedPerceptron.

Created a separate issue #1622  for adding calibration estimators in ML.NET
